### ISSUE TITLE
When Python provider raises exception, details are no longer recorded

### ISSUE THREAD

**Issue by efung** [promptfoo/promptfoo#857]
In #833 by @mldangelo , individual Python assertions that raise an exception now have their exception messages recorded as the `reason` in the `GradingResult` which is then visible as part of the set of `gradingResults` recorded for every test case when requesting JSON output.

However, an undesirable side effect of this is that if a Python **provider** raises an exception, that message is no longer visible.

Previously, say, in 0.59.1, if a provider raised an exception (e.g. incorrect OpenAI token), the exception and stack trace get dumped to the command-line.

Now, the command-line doesn't show anything, but the JSON test results now include an `error` value which begins:

```
Error: The Python script `call_api` function must return a dict with an `output` string/object or `error` string

Error: The Python script `call_api` function must return a dict with an `output` string/object or `error` string
    at PythonProvider.callApi
```

It would be nice if the exception message and stack trace were recorded in the `error` field.

**Comment by typpo** [comment]
Fix landed in #863

**Comment by typpo** [comment]
This change is released in 0.62.0
